### DISCUSSION TITLE
Is there an exhaustive list of output formats somewhere?

### DISCUSSION THREAD

**Discussion by pmaresca** [apple/pkl#145] [Q&A] [answered]
I've tried looking through the cli docs and help output but no luck. Just was wondering if
there's a current/prospective list of supported output formats.

**Comment by bioball** [comment, marked as answer]
Here's the list of output formats built into Pkl: https://pkl-lang.org/main/current/pkl-cli/index.html#format

You can define an in-language renderer, too, so there's no way to have an exhaustive list. We ourselves maintain two in-language renderers, distributed as packages:

* [toml](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/pkl.toml/current/index.html)
* [csv](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/pkl.csv/current/index.html)

**Reply by pmaresca** [reply]
Cool, so then is there a place where I can find "official" in-language renderers recognized by the pkl community?

**Reply by johnallen3d** [reply]
My guess would be the [package listing](https://pkl-lang.org/package-docs/index.html). Packages that are renderers are tagged as such.

**Reply by bioball** [reply]
Right now, no; we have a site that hosts documentation for packages maintained only by us (what @johnallen3d linked), so it wouldn't contain things from the Pkl community.

It'd be great to have some way to make community things more discoverable, but it's a larger problem that we'll need to think through how to solve.
